simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/EnablePrometheusCollectorRegistration.java

```diff
@@ -0,0 +1,52 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An {@literal @}Enable style annotation that provides CollectorRegistry related functions.
+ *
+ * <pre><code>
+ *{@literal @}Configuration
+ *{@literal @}EnablePrometheusCollectorRegistration(useDefault = true, register = true)
+ * public class MyConfiguration {
+ *
+ *  {@literal @}Bean
+ *   public Counter counter() {
```

Comment:
My recommendation/advice is to let Spring deal with this, because relationships & dependencies in Spring are best managed by the dependency injection container. It is static calls & references that are more likely to lead to bugs here.

The 'natural' way to build things in Spring is to just create the bean, and any other things that are interested in it will find & use it.

The associated BeanPostProcessor will see every Collector bean and ensure it is registered once, and once only.

Which potential bugs are you concerned about?